### ISSUE TITLE
Add 'Auto' field or something more appropriately named

### ISSUE THREAD

**Issue by rossmacarthur** [rossmacarthur/serde#111]
It would be nice to have a field that must be present in deserialization but has a default when instantiated.

Currently the way this has to be done is by overriding the `__init__` method.

```python
from serde import Model, fields

class Example(Model):
    name = fields.Str()
    
    def __init__(self, **kwargs):
        kwargs.setdefault('a', 'a default value')
        super().__init__(**kwargs)
```

**Comment by rossmacarthur** [comment]
We could also do this using a new keyword argument on the base `Field`.

**Comment by rossmacarthur** [comment]
Resolved in 63999ad5d642016a9903c45663f9c112654d62b9
